core/src/main/scala/org/http4s/client/jdkhttpclient/WebSocketClient.scala

```diff
@@ -0,0 +1,167 @@
+package org.http4s.client.jdkhttpclient
+
+import cats._
+import cats.data.Chain
+import cats.effect._
+import cats.effect.concurrent.{Deferred, Ref, TryableDeferred}
+import cats.implicits._
+import fs2.{Pipe, Pull, Stream}
+import org.http4s.{Headers, Method, Uri}
+import scodec.bits.ByteVector
+
+/**
+  * A websocket request.
+  *
+  * @param uri The URI.
+  * @param headers The headers to send. Some websocket clients reject protocol-specific headers.
+  * @param method The method of the intial HTTP request. Ignored by some clients.
+  * @param subprotocols The subprotocols, descending from most prefered.
+  */
+case class WSRequest(
```

Comment:
Is there no HTTP version in these requests?

Reply:
Isn't Websockets ONLY 1.1?

Reply:
yes, see e.g. [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism)

Reply:
actually, there are plans to support HTTP 2: [RFC](https://tools.ietf.org/html/draft-hirano-httpbis-websocket-over-http2-00)